java-sdk/README.asciidoc

```diff
@@ -0,0 +1,212 @@
+
+=== Security
+Read more on how to set up a secure environment for Java SDK connectors 
+ifdef::env-github[]
+link:security.asciidoc[here].
+endif::[]
+ifndef::env-github[]
+link:/fusion-server/{version}/search-development/getting-data-in/connectors/sdk/java-sdk/java-connector-dev.html#security[here].
+endif::[]
+
+== Java SDK Framework Overview
+Only a few components must be implemented to compile and build a functioning plugin.
+Here, we give an overview of each of these primary components and related concepts. We start by describing a minimal
+project directory layout, followed by details on the components.
+
+=== Project Layout
+This documentation and examples use Gradle for building connectors. Other build tools can be used instead.
+Check out the https://docs.gradle.org/current/userguide/tutorial_java_projects.html[Gradle quickstart^] for Java Projects.
+
+The minimal project layout is as follows:
+
+    my-connector
+    ├── build.gradle
+    ├── gradle.properties
+    ├── settings.gradle
+    └── src
+        └── main
+            └── java
+                └── Plugin.java
+                └── Config.java
+                └── Fetcher.java
+
+==== build.gradle
+This is a typical Gradle build file. A few important settings are required. You can have a look
+at an example https://github.com/lucidworks/connectors-sdk-resources/blob/master/java-sdk/connectors/build.gradle[here^].
+
+==== gradle.properties
+The build utility used for packaging up a connector depends on this file and contains required metadata about the connector,
+including its name, version, and the version of Fusion it can connect to.
+Have a look at the example https://github.com/lucidworks/connectors-sdk-resources/blob/master/java-sdk/connectors/random-connector/gradle.properties[here^].
+
+==== src/main/java
+This is where Java code lives by default. This location can be changed, but in this documentation we use the Gradle default.
+
+=== The Plugin
+The `Plugin` class describes the various subcomponents of a connector, as well as its dependencies.
+The current set of subcomponents include the `Fetcher`, validation, security filtering, and configuration suggestions.
+
+You can think of the `Plugin` class as the component that glues everything together for an implementation.
+
+You can see an example
+https://github.com/lucidworks/connectors-sdk-resources/blob/master/java-sdk/connectors/random-connector/src/main/java/com/lucidworks/fusion/connector/plugin/RandomContentPlugin.java[here^].
+Notice that each subcomponent can have a set of unique dependencies, or several subcomponents can share dependencies.
+
+=== The Configuration
+The configuration component is an interface, which extends a base `Model` interface. By simply adding `getter` methods
+and `@Property` annotations, you can dynamically generate a type-safe configuration object from a connector's configuration data.
```

Comment:
Couple of things have changed, mainly the methods no longer need to be "getters":

```
The configuration component is an interface which extends a base `Model` interface. By simply annotating methods with `@Property`, type-safe configuration objects can be generated from the interface and a connector's configuration data.
```